src/ui/pages/PageManager.ts

```diff
@@ -0,0 +1,69 @@
+import { Application } from "pixi.js";
+import { BasePage } from "./BasePage";
+
+/**
+ * The controller class to manage the page navigation.
+ */
+export class PageManager {
+    private app: Application;
+    private currentPage: BasePage;
+    private currentPageIndex: number = -1;
```

Comment:
You don't have to keep current index of a page. Page already have an `id`. you could keep that id, instead of index.

Reply:
updated as suggested;

https://github.com/irgat/soloTest/pull/3/commits/72f600defb7d8e006c8c8f7385feb04beeb6679c